pkg/operation/botanist/monitoring.go

```diff
@@ -119,6 +119,9 @@ func (b *Botanist) DeployMonitoring(ctx context.Context) error {
 		return b.Shoot.Components.Monitoring.Monitoring.Destroy(ctx)
 	}
 
+	if b.ControlPlaneWildcardCert != nil {
+		b.Operation.Shoot.Components.Monitoring.Monitoring.SetWildcardCertName(pointer.String(b.ControlPlaneWildcardCert.GetName()))
+	}
```

Comment:
There is https://github.com/gardener/gardener/blob/89debfd7475372a8f40de88804a6fe84c3085c25/pkg/operation/botanist/monitoring.go#L56-L59. Doesn't it work or?

Reply:
The only place I see `b.ControlPlaneWildcardCert` being set is https://github.com/gardener/gardener/blob/2e4617c3de1a4491a6287788266d45a7fc8044e2/pkg/operation/botanist/secrets.go#L359-L385 (invoke by the `InitializeSecretsManagement` func/step).
I rather suggest dropping https://github.com/gardener/gardener/blob/89debfd7475372a8f40de88804a6fe84c3085c25/pkg/operation/botanist/monitoring.go#L56-L59 and similar handlings as according to my understanding when `DefaultMonitoring` is invoked, then `b.ControlPlaneWildcardCert` will be always nil.